src/dependency_specifications.rb

```diff
@@ -0,0 +1,31 @@
+
+# frozen_string_literal: true
+
+module Metalware
+  # Class to generate reused `dependency_hash`s, for use by `Dependency` object
+  # to enforce command dependencies.
+  # XXX Consider moving generating of more `dependency_hash`s here.
+  class DependencySpecifications
+    def initialize(config)
+      @config = config
+    end
+
+    def for_node_in_configured_group(node_name)
+      primary_group = configured_primary_group_for_node(node_name)
+      {
+        repo: ['configure.yaml'],
+        configure: ['domain.yaml', "groups/#{primary_group}.yaml"],
```

Comment:
I think this also needs `nodes/#{node}.yaml` to insure that the node answers are validated. We could possible use this for validating the configure commands.

It might be also useful to include a group option